File: windows/Liferay.SDK/Service/V73/Commerceaccount/CommerceaccountService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commerceaccount
{
	public class CommerceaccountService : ServiceBase
	{
		public CommerceaccountService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> SetActiveAsync(long commerceAccountId, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/set-active", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceAccountAsync(long commerceAccountId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/get-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddBusinessCommerceAccountAsync(string name, long parentCommerceAccountId, string email, string taxId, bool active, string externalReferenceCode, IEnumerable<long> userIds, IEnumerable<string> emailAddresses, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("name", name);
			_parameters.Add("parentCommerceAccountId", parentCommerceAccountId);
			_parameters.Add("email", email);
			_parameters.Add("taxId", taxId);
			_parameters.Add("active", active);
			_parameters.Add("externalReferenceCode", externalReferenceCode);
			_parameters.Add("userIds", userIds);
			_parameters.Add("emailAddresses", emailAddresses);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/add-business-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchByExternalReferenceCodeAsync(long companyId, string externalReferenceCode)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("externalReferenceCode", externalReferenceCode);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/fetch-by-external-reference-code", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceAccountAsync(string name, long parentCommerceAccountId, string email, string taxId, int type, bool active, string externalReferenceCode, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("name", name);
			_parameters.Add("parentCommerceAccountId", parentCommerceAccountId);
			_parameters.Add("email", email);
			_parameters.Add("taxId", taxId);
			_parameters.Add("type", type);
			_parameters.Add("active", active);
			_parameters.Add("externalReferenceCode", externalReferenceCode);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/add-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceAccountAsync(long commerceAccountId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/delete-commerce-account", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> FetchCommerceAccountAsync(long commerceAccountId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/fetch-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetPersonalCommerceAccountAsync(long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/get-personal-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserCommerceAccountsAsync(long userId, long parentCommerceAccountId, int commerceSiteType, string keywords, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("parentCommerceAccountId", parentCommerceAccountId);
			_parameters.Add("commerceSiteType", commerceSiteType);
			_parameters.Add("keywords", keywords);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/get-user-commerce-accounts", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserCommerceAccountsAsync(long userId, long parentCommerceAccountId, int commerceSiteType, string keywords, bool active, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("parentCommerceAccountId", parentCommerceAccountId);
			_parameters.Add("commerceSiteType", commerceSiteType);
			_parameters.Add("keywords", keywords);
			_parameters.Add("active", active);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/get-user-commerce-accounts", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetUserCommerceAccountsCountAsync(long userId, long parentCommerceAccountId, int commerceSiteType, string keywords)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("parentCommerceAccountId", parentCommerceAccountId);
			_parameters.Add("commerceSiteType", commerceSiteType);
			_parameters.Add("keywords", keywords);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/get-user-commerce-accounts-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetUserCommerceAccountsCountAsync(long userId, long parentCommerceAccountId, int commerceSiteType, string keywords, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("parentCommerceAccountId", parentCommerceAccountId);
			_parameters.Add("commerceSiteType", commerceSiteType);
			_parameters.Add("keywords", keywords);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/get-user-commerce-accounts-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCommerceAccountAsync(long commerceAccountId, string name, bool logo, byte[] logoBytes, string email, string taxId, bool active, long defaultBillingAddressId, long defaultShippingAddressId, string externalReferenceCode, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);
			_parameters.Add("name", name);
			_parameters.Add("logo", logo);
			_parameters.Add("logoBytes", logoBytes);
			_parameters.Add("email", email);
			_parameters.Add("taxId", taxId);
			_parameters.Add("active", active);
			_parameters.Add("defaultBillingAddressId", defaultBillingAddressId);
			_parameters.Add("defaultShippingAddressId", defaultShippingAddressId);
			_parameters.Add("externalReferenceCode", externalReferenceCode);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/update-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCommerceAccountAsync(long commerceAccountId, string name, bool logo, byte[] logoBytes, string email, string taxId, bool active, long defaultBillingAddressId, long defaultShippingAddressId, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);
			_parameters.Add("name", name);
			_parameters.Add("logo", logo);
			_parameters.Add("logoBytes", logoBytes);
			_parameters.Add("email", email);
			_parameters.Add("taxId", taxId);
			_parameters.Add("active", active);
			_parameters.Add("defaultBillingAddressId", defaultBillingAddressId);
			_parameters.Add("defaultShippingAddressId", defaultShippingAddressId);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/update-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCommerceAccountAsync(long commerceAccountId, string name, bool logo, byte[] logoBytes, string email, string taxId, bool active, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);
			_parameters.Add("name", name);
			_parameters.Add("logo", logo);
			_parameters.Add("logoBytes", logoBytes);
			_parameters.Add("email", email);
			_parameters.Add("taxId", taxId);
			_parameters.Add("active", active);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/update-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateDefaultBillingAddressAsync(long commerceAccountId, long commerceAddressId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);
			_parameters.Add("commerceAddressId", commerceAddressId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/update-default-billing-address", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateDefaultShippingAddressAsync(long commerceAccountId, long commerceAddressId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountId", commerceAccountId);
			_parameters.Add("commerceAddressId", commerceAddressId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/update-default-shipping-address", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpsertCommerceAccountAsync(string name, long parentCommerceAccountId, bool logo, byte[] logoBytes, string email, string taxId, int type, bool active, string externalReferenceCode, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("name", name);
			_parameters.Add("parentCommerceAccountId", parentCommerceAccountId);
			_parameters.Add("logo", logo);
			_parameters.Add("logoBytes", logoBytes);
			_parameters.Add("email", email);
			_parameters.Add("taxId", taxId);
			_parameters.Add("type", type);
			_parameters.Add("active", active);
			_parameters.Add("externalReferenceCode", externalReferenceCode);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccount/upsert-commerce-account", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commerceaccount/CommerceaccountService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commerceaccount;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommerceaccountService extends BaseService {

	public CommerceaccountService(Session session) {
		super(session);
	}

	public JSONObject addBusinessCommerceAccount(String name, long parentCommerceAccountId, String email, String taxId, boolean active, String externalReferenceCode, JSONArray userIds, JSONArray emailAddresses, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("email", checkNull(email));
			_params.put("emailAddresses", checkNull(emailAddresses));
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));
			_params.put("name", checkNull(name));
			_params.put("parentCommerceAccountId", parentCommerceAccountId);
			_params.put("taxId", checkNull(taxId));
			_params.put("userIds", checkNull(userIds));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceaccount/add-business-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject addCommerceAccount(String name, long parentCommerceAccountId, String email, String taxId, int type, boolean active, String externalReferenceCode, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("email", checkNull(email));
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));
			_params.put("name", checkNull(name));
			_params.put("parentCommerceAccountId", parentCommerceAccountId);
			_params.put("taxId", checkNull(taxId));
			_params.put("type", type);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceaccount/add-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceAccount(long commerceAccountId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountId", commerceAccountId);

			_command.put("/commerce.commerceaccount/delete-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject fetchByExternalReferenceCode(long companyId, String externalReferenceCode) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));

			_command.put("/commerce.commerceaccount/fetch-by-external-reference-code", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject fetchCommerceAccount(long commerceAccountId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountId", commerceAccountId);

			_command.put("/commerce.commerceaccount/fetch-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getCommerceAccount(long commerceAccountId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountId", commerceAccountId);

			_command.put("/commerce.commerceaccount/get-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getPersonalCommerceAccount(long userId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("userId", userId);

			_command.put("/commerce.commerceaccount/get-personal-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getUserCommerceAccounts(long userId, long parentCommerceAccountId, int commerceSiteType, String keywords, boolean active, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceSiteType", commerceSiteType);
			_params.put("end", end);
			_params.put("keywords", checkNull(keywords));
			_params.put("parentCommerceAccountId", parentCommerceAccountId);
			_params.put("start", start);
			_params.put("userId", userId);

			_command.put("/commerce.commerceaccount/get-user-commerce-accounts", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getUserCommerceAccounts(long userId, long parentCommerceAccountId, int commerceSiteType, String keywords, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceSiteType", commerceSiteType);
			_params.put("end", end);
			_params.put("keywords", checkNull(keywords));
			_params.put("parentCommerceAccountId", parentCommerceAccountId);
			_params.put("start", start);
			_params.put("userId", userId);

			_command.put("/commerce.commerceaccount/get-user-commerce-accounts", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getUserCommerceAccountsCount(long userId, long parentCommerceAccountId, int commerceSiteType, String keywords) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceSiteType", commerceSiteType);
			_params.put("keywords", checkNull(keywords));
			_params.put("parentCommerceAccountId", parentCommerceAccountId);
			_params.put("userId", userId);

			_command.put("/commerce.commerceaccount/get-user-commerce-accounts-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getUserCommerceAccountsCount(long userId, long parentCommerceAccountId, int commerceSiteType, String keywords, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceSiteType", commerceSiteType);
			_params.put("keywords", checkNull(keywords));
			_params.put("parentCommerceAccountId", parentCommerceAccountId);
			_params.put("userId", userId);

			_command.put("/commerce.commerceaccount/get-user-commerce-accounts-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject setActive(long commerceAccountId, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceAccountId", commerceAccountId);

			_command.put("/commerce.commerceaccount/set-active", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceAccount(long commerceAccountId, String name, boolean logo, byte[] logoBytes, String email, String taxId, boolean active, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceAccountId", commerceAccountId);
			_params.put("email", checkNull(email));
			_params.put("logo", logo);
			_params.put("logoBytes", toString(logoBytes));
			_params.put("name", checkNull(name));
			_params.put("taxId", checkNull(taxId));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceaccount/update-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceAccount(long commerceAccountId, String name, boolean logo, byte[] logoBytes, String email, String taxId, boolean active, long defaultBillingAddressId, long defaultShippingAddressId, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceAccountId", commerceAccountId);
			_params.put("defaultBillingAddressId", defaultBillingAddressId);
			_params.put("defaultShippingAddressId", defaultShippingAddressId);
			_params.put("email", checkNull(email));
			_params.put("logo", logo);
			_params.put("logoBytes", toString(logoBytes));
			_params.put("name", checkNull(name));
			_params.put("taxId", checkNull(taxId));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceaccount/update-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceAccount(long commerceAccountId, String name, boolean logo, byte[] logoBytes, String email, String taxId, boolean active, long defaultBillingAddressId, long defaultShippingAddressId, String externalReferenceCode, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceAccountId", commerceAccountId);
			_params.put("defaultBillingAddressId", defaultBillingAddressId);
			_params.put("defaultShippingAddressId", defaultShippingAddressId);
			_params.put("email", checkNull(email));
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));
			_params.put("logo", logo);
			_params.put("logoBytes", toString(logoBytes));
			_params.put("name", checkNull(name));
			_params.put("taxId", checkNull(taxId));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceaccount/update-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateDefaultBillingAddress(long commerceAccountId, long commerceAddressId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountId", commerceAccountId);
			_params.put("commerceAddressId", commerceAddressId);

			_command.put("/commerce.commerceaccount/update-default-billing-address", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateDefaultShippingAddress(long commerceAccountId, long commerceAddressId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountId", commerceAccountId);
			_params.put("commerceAddressId", commerceAddressId);

			_command.put("/commerce.commerceaccount/update-default-shipping-address", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject upsertCommerceAccount(String name, long parentCommerceAccountId, boolean logo, byte[] logoBytes, String email, String taxId, int type, boolean active, String externalReferenceCode, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("email", checkNull(email));
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));
			_params.put("logo", logo);
			_params.put("logoBytes", toString(logoBytes));
			_params.put("name", checkNull(name));
			_params.put("parentCommerceAccountId", parentCommerceAccountId);
			_params.put("taxId", checkNull(taxId));
			_params.put("type", type);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceaccount/upsert-commerce-account", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}